### PR TITLE
feat: adding a PCA aggregator

### PR DESCRIPTION
**Pull request by chanind**
This PR adds a PCA aggregator, usable like below:

```
from steering_vectors import pca_aggregator, train_steering_vector

vec = train_steering_vector(model, tokenizer, data, aggregator=pca_aggregator)
```

### CODE REVIEW

**Comment by dtch1997** [comment]
@chanind great work! Is the implementation a match for what they're doing in RepE?

**Comment by chanind** [comment]
It's close, but not identical to RepE's implementation. In RepE, there's the following differences:

### Scikit-learn vs PyTorch
they use scikit-learn for PCA instead of pytorch (not sure why - maybe scikit learn's implementation is better?). I didn't want to do that since it would require adding another dependency to the project, and scikit-learn is a big dependency. It wouldn't be a crazy thing to add though, so if there's a compelling reason why scikit-learn's PCA is better than just using SVD in PyTorch we should do it

### Inverting deltas
In RepE, they essentially flip deltas at random by flipping training samples at random. I don't think this is the correct thing to do for PCA. Here, we instead include both the flipped and non-flipped deltas and pass that too PCA. This has the advantage of not requiring further centering the data, and will not error for a single delta like RepE will. In addition, I feel this is more mathematically sound than what RepE is doing.

### Sign calculation
Both here and in RepE, we try to figure out if the PCA vec needs to be flipped. I think our logic is similar to how we figure this out, but haven't completely scrutinized the code in RepE to ensure it's 100% identical. The RepE code is here: https://github.com/andyzoujm/representation-engineering/blob/main/repe/rep_readers.py#L87C2-L95C98

**Comment by dtch1997** [comment]
Fair enough! I think it'd be ideal if we could also reproduce some results from RepE using this codebase, but perhaps that's out of scope of this PR. 

LGTM

**Review by dtch1997 — APPROVED** [review]
